runtime/vm/callin.cpp

```diff
@@ -1341,6 +1344,12 @@ jitFillOSRBuffer(struct J9VMThread *currentThread, void *osrBlock)
 void JNICALL
 initializeAttachedThread(J9VMThread *currentThread, const char *name, j9object_t *group, UDATA daemon, J9VMThread *initializee)
 {
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+	J9JavaVM *vm = currentThread->javaVM;
+	if (VM_CRIUHelpers::isJVMInSingleThreadMode(vm) && VM_VMHelpers::threadCanRunJavaCode(currentThread)) {
```

Comment:
Is there any race condition here? i.e.

T1 goes into single thread mode
T2 performs the test above
T1 finishes single thread mode, unmarking all threads
T2 marks itself

Reply:
What is the actual situation you're trying to prevent? Why would a thread exist but not be marked by going into single thread mode?

Wondering if this code should actually be in `allocateVMThread` instead along side the exclusive check:
https://github.com/eclipse-openj9/openj9/blob/96b1e95e98e41d9841a278b8d5931f5cf68219d0/runtime/vm/vmthread.c#L258-L270

Reply:
> What is the actual situation you're trying to prevent? Why would a thread exist but not be marked by going into single thread mode?

We are seeing cases where the finalizer thread attaches while we are already in single thread mode, so it is not in the thread list when we halt all threads. Its an intermittent issue, very timing dependent.

There may be other cases where this may occur.